scripts: kconfig: Add min/max functions

pdgendt opened the pull request:
Similar to arithmetic functions, min/max can be useful to use in kconfig.

pdgendt committed:
tests: kconfig: functions: Fix arithmetic typo

Moved the test to a single main file, fixed a typo
arithmetric -> arithmetic

Signed-off-by: Pieter De Gendt <pieter.degendt@basalte.be>

pdgendt committed:
scripts: kconfig: Add min/max functions

Similar to arithmetic functions, min/max can be useful to use in kconfig.

Signed-off-by: Pieter De Gendt <pieter.degendt@basalte.be>

pdgendt committed:
tests: kconfig: functions: Add min/max functions

Add test case for kconfig min/max functions.

Signed-off-by: Pieter De Gendt <pieter.degendt@basalte.be>

henrikbrixandersen commented:
We need to have all of these arithmetic functions documented in https://docs.zephyrproject.org/latest/build/kconfig/preprocessor-functions.html

pdgendt committed:
doc: build: kconfig: Update preprocessor functions

Split preprocessor functions into different categories and add newly
introduced functions.

Signed-off-by: Pieter De Gendt <pieter.degendt@basalte.be>

nordicjm commented on doc/build/kconfig/preprocessor-functions.rst:
```diff
@@ -75,11 +74,49 @@ while the ``*_hex`` version returns a hexadecimal value starting with ``0x``.
+
+.. code-block:: none
+
+   $(add,<value>[,value]+)
+   $(sub,<value>[,value]+)
+   $(mul,<value>[,value]+)
+   $(div,<value>[,value]+)
+   $(mod,<value>[,value]+)
+   $(inc,<value>[,value]+)
+   $(dec,<value>[,value]+)
+   $(min,<value>[,value]+)
+   $(max,<value>[,value]+)
```
would use ... not +

pdgendt replied:
Updated

kartben commented on doc/build/kconfig/preprocessor-functions.rst:
```diff
@@ -75,11 +74,49 @@ while the ``*_hex`` version returns a hexadecimal value starting with ``0x``.
+Integer functions
+*****************
+
+The functions listed below can be used to do arithmetic operations
+on integer variables, such as addition, subtraction and more.
```
it feels like the blurb from kconfigfunctions.py (" if there is only one argument, it will be split with a comma and processed as a sequence of number") would be useful to have here? I was originally confused that there was only 1 required `value` for functions such as add and thought this was a mistake

pdgendt replied:
It does state "See the Python docstrings in [scripts/kconfig/kconfigfunctions.py](https://github.com/zephyrproject-rtos/zephyr/blob/main/scripts/kconfig/kconfigfunctions.py) for detailed documentation." 😆
